src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -48,11 +71,47 @@ public String getIndentLevelString(final int planNumber, final int planLevel, fi
 	 */
 	public void planWalk( final LogicalPlan plan, final int planNumber, final int planLevel, final int numberOfSiblings, final PrintStream out) {
 		final String indentLevelString = getIndentLevelString(planNumber, planLevel, numberOfSiblings);
-		out.append( indentLevelString + plan.getRootOperator().toString() );
-		out.append( System.lineSeparator() );
+		final LogicalOperator rootOp = plan.getRootOperator();
+		final String indentLevelStringForOpDetail = getIndentLevelStringForDetail(planNumber, planLevel, numberOfSiblings);
```

Comment:
For better code readability, switch the order of these two lines (because, then, `indentLevelStringForOpDetail` is introduced directly below `indentLevelString`).